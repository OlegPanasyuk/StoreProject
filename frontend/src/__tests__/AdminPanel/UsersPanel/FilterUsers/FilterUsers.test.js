import React from 'react';
import FilterUsersConnected, { FilterUsers } from '../../../../AdminPanel/UsersPanel/FilterUsers';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
   
    updateState: ()=>{}
}) {
    let component = shallow(<FilterUsers {...props} />);
    return { component, props };
}

describe('FilterUsers --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(FilterUsers.propTypes.updateState).toBe(PropTypes.func);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});


describe('FilterUsers ', () => {
    let initialState = {
        
    };

    const mockStore = configureStore();
    let store, wrapper;
    
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><FilterUsersConnected /></Provider>);
    });
    it('+++ render the connected(SMART) component',() => {
        expect(wrapper.find(FilterUsersConnected).length).toEqual(1);
    });
});

