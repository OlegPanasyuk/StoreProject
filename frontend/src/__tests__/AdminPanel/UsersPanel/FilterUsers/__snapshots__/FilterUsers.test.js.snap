// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterUsers --- Snapshot Component must be rendered 1`] = `
<Form
  as="form"
  inline={false}
>
  <FormGroup
    as="div"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Role:
    </FormLabel>
    <FormControl
      as="input"
      onChange={[Function]}
      placeholder="text input for search name"
      type="text"
    />
  </FormGroup>
  <FormGroup
    as="div"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Role:
    </FormLabel>
    <FormControl
      as="input"
      onChange={[Function]}
      placeholder="User"
      type="text"
    />
  </FormGroup>
</Form>
`;
